src/ensembl/src/content/app/entity-viewer/components/transcript-filter/EntityViewerTranscriptFilter.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+import styles from './EntityViewerTranscriptFilter.scss';
+
+export type Option = {
+  value: string;
+  label: string;
+};
+
+export type OptionsGroup = Option[];
+
+type Props = {
+  selectedValues: string[];
+  onChange: (selectedValues: string[]) => void;
+  options: OptionsGroup[];
+  hideUnchecked?: boolean;
+};
+
+const EntityViewerTranscriptFilter = (props: Props) => {
+  const handleChange = (status: boolean, optionIndex: [number, number]) => {
+    const [groupIndex, itemIndex] = optionIndex;
+    const selectedOption = props.options[groupIndex][itemIndex].value;
+
+    if (status) {
+      const newSelectedOptions = [...props.selectedValues, selectedOption];
+      props.onChange(newSelectedOptions);
+    } else {
+      const newSelectedOptions = [...props.selectedValues].filter(
+        (option) => option != selectedOption
+      );
+
+      props.onChange(newSelectedOptions);
+    }
+  };
+
+  return (
+    <div className={styles.defaultWrapper}>
+      {props.options.map((options, groupIndex) => {
+        const optionGroupClassName = classNames(styles.checkboxPadding, {
+          [styles.optionGroup]: !props.hideUnchecked
```

Comment:
If we remove the hideUnchecked prop, then the `checkboxPadding` class should be renamed to optionGroup as better reflecting what it is for

Reply:
I guess this is not necessary anymore?